.github/workflows/release.yml

```diff
@@ -21,12 +21,11 @@ jobs:
         with:
           fetch-depth: 0
       - name: Calculate go version
-        id: vars
-        run: echo "go_version=$(make go-version)" >> $GITHUB_OUTPUT
+        run: echo "go_version=$(make go-version)" >> $GITHUB_ENV
```

Comment:
Are we relying upon the ubuntu OS to determine which version of go we're using here? How about using a specific action that consumed the version of go declared in `go.mod`, e.g.:

```yaml
      - name: Install go
        uses: actions/setup-go@v3
        with:
          go-version-file: 'go.mod'
```

Reply:
This is the go version - including the patch version - set in the Makefile. It's the same version that informs the `release-binaries` make target.

Reply:
Got it. If the makefile consumes go.mod then that's great otherwise, we probably wanna be aware that we're declaring a go version in > 1 place which is probably not ideal.

In any event, not something to worry about in this PR in any case!

Reply:
I think there's a difference between the build version of `go` - in the Makefile - and the module version, so they should be declared separately. We're definitely doing that currently in CAPI. The build version is v1.19.6 on main currently, while the module version is 1.19.

I think currently we're declaring the go version in 5 places - 3 X go.mod + Makefile + golangci-lint.yml.

Reply:
Please see: https://github.com/kubernetes-sigs/cluster-api/issues/7974#issuecomment-1403150288

Simply grabing the major.minor version from go.mod file is not good enough if we want repeatable builds (i.e. a pinned patch version)

Reply:
Yep, ACK!